src/Npgsql/ConnectorPool.cs

```diff
@@ -387,7 +387,7 @@ await Task.WhenAny(tcs.Task, Task.Delay(timeLeft, delayCancellationToken.Token))
                                         throw new NpgsqlException($"The connection pool has been exhausted, either raise MaxPoolSize (currently {_max}) or Timeout (currently {Settings.Timeout} seconds)");
                                 }
                                 else
-                                    tcs.Task.Wait();
+                                    await tcs.Task;
```

Comment:
This code is intentionally synchronous and should not await - we are in a condition that behaves differently in async vs. sync.